### ISSUE TITLE
When using the "Shrinking" option to shrink an existing partition to be able to create a new one, in the new partition allow you to select the type of partition you want to create since it only allows you to create Ext4 type

### ISSUE THREAD

**Issue by Alden20** [calamares/calamares#2114]
**Is your feature request related to a problem? Please describe.**

I have an existing partition of type Btrfs with a Linux installed that occupies the entire disk. Now I want to create another partition, also of type Btrfs, to install another Linux on it, so that I have two Linuxes installed on the disk, each on a Btrfs partition.

To do this, I use the Shrinking option to shrink the existing partition and leave space to create the new one without losing the content of the existing partition, thus dividing the space of the partition in half. The problem is that when I select this option it does not give me any option to be able to select the type of the new partition that is going to be created, always establishing by default that it is of type Ext4 without giving me any option to be able to change it, with which I am forced to have the original partition of type Btrfs and the new partition that I am going to create of type Ext4.

**Describe the solution you'd like**

Please, I wanted to ask if it would be possible to add the possibility of being able to choose in this case the type of the new partition to be created, in the same way that when you choose to install Linux using all the disk space you are given the option to choose the type of partition to be created (Ext4, Btrfs, XFS...). I think it is the most logical procedure, that the user is allowed, at the time of shrinking the disk, to choose what type of partition he wants the new one to be. I also think that it would be a functionality very well received by users.

**Describe alternatives you've considered**

I imagine that maybe it is possible to use some external application before running Calamares to be able to shrink the existing partition of type Btrfs, resize it, create another new partition of type Btrfs and tell Calamares to install Linux there using that partition. People have recommended me to use the KDE Partition Manager to do this before launching Calamares, I don't know because I haven't been able to test it.

But from my point of view, I think that this is a basic functionality that should be offered directly by Calamares without having to be forced to use an external application, it seems completely logical and normal to me in the installation process that when choosing the Shrinking option it allows you to choose the type of the new partition to created, I don't understand why when shrinking the existing partition, the new partition has to be created as Ext4 without giving the user an option to choose what type of partition he wants to create, I think it doesn't exist no technical problem for it since different types of Linux partitions can coexist on the same disk.

**Comment by demmm** [comment]
Are you a distribution developer or an end user?  If the latter it might be better to first discuss with your distribution:
https://calamares.io/issues/
https://calamares.io/issues/#end-users

If a distro dev, is this tested with the automated install options or manual? What version of calamares in use?  Logs will be needed too.
https://calamares.io/issues/
https://calamares.io/issues/#developers-of-distributions

**Comment by Alden20** [comment]
I'm an end user. I'm using Manjaro (currently Sikaris 22.0.4).

I have first discussed this problem on the Manjaro forum:

https://forum.manjaro.org/t/shrinking-a-btrfs-partition-from-calamares/136440

But from there they have recommended me to make a feature request to Calamares:

"You can file a feature request to calamares.io and they might follow your line of thought. But their reasoning is that ext4 is default and for other filesystem options use manual partitioning."

**Comment by demmm** [comment]
>But their reasoning is that ext4 is default and for other filesystem options use manual partitioning.

Setting a default for filesystem type is up to a distribution, Manjaro can set whatever they want as default (including BTRFS), but that is not the point here anyway.
https://github.com/calamares/calamares/blob/calamares/src/modules/partition/partition.conf#L169

**Comment by Alden20** [comment]
Sorry for my ignorance, but is this valid when the Shrinking option is selected?

Looking at the configuration file that you have indicated here, I see this comment:

# When erasing the disk, the *defaultFileSystemType* is used (see
# above), but it is also possible to give users a choice:
# list suitable filesystems here. A drop-down is provided
# to pick which is the filesystems will be used.

But I understand that in this case it is referring to when the entire disk is erased, not when the Shrinking option is used.

In fact, in Manjaro's Calamares, when you select to erase and use the entire disk for the Linux installation, it does allow you to choose the type of partition you want to create.

But reading that comment, would this option also be valid when using the Shrinking option? Would a dropdown list to select the new partition type also appear when using this option? Or would it only be valid when the entire drive is erased?

**Comment by dalto8** [comment]
Currently Calamares allows choosing the filesystem when using both the "Erase Disk" option and the "Replace Partition" option.  The only time we don't offer the choice is when using the "Install Alongside" option which does seem like a feature gap.

**Comment by Alden20** [comment]
That's what I'm trying to say. When the "Install alongside" option is selected, it does not allow you to select the type of the new partition to be created, and forces you to always create it by default as Ext4 type without having any possibility of changing it or being able to select it from a list in the same way that is done in the "Erase Disk" and "Replace Partition" options.

As you say, dalto8, I think it's a feature gap, and that's why I'm asking if maybe it would be possible to implement this in a future version.

I don't know if technically it would be too difficult to implement or if it would be a big change in the application, but it would simply be to use the list of filesystem types that already came in the configuration file and are being used right now for the options "Erase Disk" and "Replace Partition", it would show the same drop-down list and allow you to select the partition type.

**Comment by demmm** [comment]
> When the "Install alongside" option is selected, it does not allow you to select the type of the new partition to be created

That is a duplicate of https://github.com/calamares/calamares/issues/1911 & https://github.com/calamares/calamares/issues/2019
That is why I asked for calamares version, since that feature was merged last December.

So to be clear, you are asking to always have filesystem selection option?  Or are you reporting a bug, that the option to select any is gone when a BTRFS partition is used for shrinking?

**Comment by dalto8** [comment]
> That is a duplicate of #1911 & #2019

#1911 was only about "Replace Partition" which is implemented.  #2019 was closed as a duplicate of #1911 even though "Install Alongside" was never addressed.  Either this one, which is specific to "Install Alongside" or #2019 which was only partially implemented should stay open.

**Comment by demmm** [comment]
@dalto still not a clear answer if this is about a bug with BTRFS (regardless whether alongside has a dropdown menu), if at any time you select to shrink BTRFS (automated or manual), removes any option to set anything but ext4, then this is not a feature request, but a bug.

**Comment by dalto8** [comment]
> still not a clear answer if this is about a bug with BTRFS (regardless whether alongside has a dropdown menu), if at any time you select to shrink BTRFS (automated or manual), removes any option to set anything but ext4, then this is not a feature request, but a bug.

I don't understand your point.  When you shrink using the "Install Alongside" option there is no way to choose a filesystem.  As a result, there is no option to begin with.  There can't be a bug that causes the option to be removed because there was never an option to begin with.

**Comment by demmm** [comment]
>if at any time you select to shrink BTRFS (automated or manual)

Selection should be there with manual.

**Comment by kkofler** [comment]
But this bug report is about having the selection also with automatic "Install Alongside" partitioning.

**Comment by demmm** [comment]
No-where in the first post is it mentioned that automated is used, only mentions "when shrinking", that is why so many times I have asked now, is this also the case for manual, or automated only.

**Comment by kkofler** [comment]
The subsequent comments have made it very clear that we are talking about the "Install Alongside" option here, i.e., one of the 3 automatic partitioning options.

We all know that the partition type can already be selected for the 2 other automatic options and of course for manual partitioning (where everything can be selected).

**Comment by Alden20** [comment]
I am currently using the latest version of Manjaro (Sikaris 22.0.4) which uses Calamares 3.2.61-10.

Sorry, maybe I didn't explain myself well at the beginning. What I am trying to say is that when the "Install Alongside" option is selected, it is not allowed at any time to be able to choose the type of the new partition that is going to be created after shrinking, by default the system forces it to be type Ext4 without give no chance to change it in the process.

If on my disk I already have a Btrfs partition created with Linux that occupies the entire disk and now I want to install another Linux on another partition of type Btrfs and for this I try to use the "Install Alongside" option to divide the drive space in two partitions keeping the data of the first partition, I can't do this because the application forces by default that the new partition to be created is of type Ext4, without giving me any option to be able to select the type I want or change it during the process .

I'm trying to say that, from my point of view, this is a lack of functionality, it is allowed to choose the type of partition you want to create in the rest of the options ("Erase Disk" and "Replace Partition"), but it is not allowed in the "Install Alongside" option, which is illogical from my point of view.

**Comment by D3vil0p3r** [comment]
> Currently Calamares allows choosing the filesystem when using both the "Erase Disk" option and the "Replace Partition" option. The only time we don't offer the choice is when using the "Install Alongside" option which does seem like a feature gap.

Hey mate. I was looking for "Replace Partition" but I don't see it with file system choice... only Erase Disk has it.

**Comment by dalto8** [comment]
> Hey mate. I was looking for "Replace Partition" but I don't see it with file system choice... only Erase Disk has it.

That change is only in the latest code.  Are you using that version?

**Comment by D3vil0p3r** [comment]
> > Hey mate. I was looking for "Replace Partition" but I don't see it with file system choice... only Erase Disk has it.
> 
> That change is only in the latest code. Are you using that version?

I'm using the latest version of Calamares, not sure if I'm using one of the latest git commit.

**Comment by dalto8** [comment]
> I'm using the latest version of Calamares, not sure if I'm using one of the latest git commit.

Well, what commit did you build off of?

**Comment by D3vil0p3r** [comment]
> > I'm using the latest version of Calamares, not sure if I'm using one of the latest git commit.
> 
> Well, what commit did you build off of?

I don't remember but it should be one of the latest commits. By the way, why the choice to remove the File System selection from "Replace Partition"?

**Comment by D3vil0p3r** [comment]
If the file system choice was 'never' there (Replace Partition), what is the meaning of:

> Currently Calamares allows choosing the filesystem when using both the "Erase Disk" option and the "Replace Partition" option. The only time we don't offer the choice is when using the "Install Alongside" option which does seem like a feature gap.

**Comment by D3vil0p3r** [comment]
By the way, why the choice to not have the File System choice on Install alongside and Replace partition but only on Erase disk?
